packages/cli/src/evaluation/test-definitions.controller.ee.ts

```diff
@@ -0,0 +1,87 @@
+import { Get, Post, Patch, RestController, Delete } from '@/decorators';
+import { BadRequestError } from '@/errors/response-errors/bad-request.error';
+import { NotFoundError } from '@/errors/response-errors/not-found.error';
+import { listQueryMiddleware } from '@/middlewares';
+import { getSharedWorkflowIds } from '@/public-api/v1/handlers/workflows/workflows.service';
+import { isPositiveInteger } from '@/utils';
+
+import { TestDefinitionsService } from './test-definitions.service.ee';
+import { TestDefinitionsRequest } from './test-definitions.types.ee';
+
+@RestController('/evaluation/test-definitions')
+export class TestDefinitionsController {
+	constructor(private readonly testsService: TestDefinitionsService) {}
+
+	@Get('/', { middlewares: listQueryMiddleware })
+	async getMany(req: TestDefinitionsRequest.GetMany) {
+		const workflowIds = await getSharedWorkflowIds(req.user, ['workflow:read']);
+
+		return await this.testsService.getMany(req.listQueryOptions, workflowIds);
+	}
+
+	@Get('/:id')
+	async getOne(req: TestDefinitionsRequest.GetOne) {
+		if (!isPositiveInteger(req.params.id)) {
+			throw new BadRequestError('Test ID is not a number');
+		}
+
+		const workflowIds = await getSharedWorkflowIds(req.user, ['workflow:read']);
+
+		const testDefinition = await this.testsService.findOne(Number(req.params.id), workflowIds);
+
+		if (!testDefinition) throw new NotFoundError('Test not found');
+
+		return testDefinition;
+	}
+
+	@Post('/')
+	async create(req: TestDefinitionsRequest.Create) {
+		const workflowIds = await getSharedWorkflowIds(req.user, ['workflow:read']);
+
+		if (!workflowIds.includes(req.body.workflowId)) {
+			throw new BadRequestError('User does not have access to the workflow');
+		}
+
+		if (req.body.evaluationWorkflowId && !workflowIds.includes(req.body.evaluationWorkflowId)) {
```

Comment:
Does this mean you have to have an existing evaluation workflow before creating a test definition? Isn't that restrictive? When the process get's interrupted (by coffee, whatever) when creating a test, does that mean the work is not saved until the evaluation workflow is selected / created?

Reply:
oh, woops, thanks for the clarification!